packages/agreement-process/src/model/domain/validators.ts

```diff
@@ -69,6 +69,26 @@ export const assertRequesterIsConsumer = (
   }
 };
 
+export function assertRequesterIsProducer(
+  requesterId: string,
+  agreement: Agreement
+): void {
+  if (agreement.producerId !== requesterId) {
+    throw operationNotAllowed(requesterId);
+  }
+}
+
+export const assertRequesterIsConsumerOrProducer = (
+  requesterId: string,
+  agreement: Agreement
+): void => {
+  try {
+    assertRequesterIsConsumer(requesterId, agreement.consumerId);
```

Comment:
may we use the same signature as `assertRequesterIsProducer` and take an agreement instead of just the id?
(doing it in another PR is fine)